main.cpp

```diff
@@ -127,17 +155,152 @@ struct IntType
             heapInt = nullptr;
         }
 
-        operator int() { return *heapInt; }
+        operator int() const { return *heapInt; }
 
         IntType& add( int operand );
         IntType& subtract( int operand );
         IntType& multiply( int operand );  
         IntType& divide( int operand );
+
+        IntType& pow( const FloatType& operand );
+        IntType& pow( const DoubleType& operand );
+        IntType& pow( const IntType& operand );
+        IntType& pow( int operand );
     
     private:
         int* heapInt = nullptr;
+        IntType& powInternal( int x );
 };
 
+//point constructor implementations
+Point::Point(FloatType& a, FloatType& b) : x(a), y(b) {}
+
+Point::Point(DoubleType& a, DoubleType& b) : x( static_cast<float>(a) ), y( static_cast<float>(b) ) {}
+
+Point::Point(IntType& a, IntType& b) : x( static_cast<float>(a) ), y( static_cast<float>(b) ) {}
```

Comment:
All of these should call `Point(float, float)`.  You can look up Forwarding Constructors if you want to learn more: 
```
Point::Point(FloatType& a, FloatType& b) : Point(static_cast<float>(a), static_cast<float>(b) ) { } 
``` 
- [ ] revise